unittests/test_policies.py

```diff
@@ -722,8 +728,8 @@ def test_kbd_interrupt_in_wait_with_limited_concurrency(
     runner, _ = async_runner
     with pytest.raises(KeyboardInterrupt):
         runner.runall(make_cases([
-            KeyboardInterruptCheck(), SleepCheck(10),
-            SleepCheck(10), SleepCheck(10)
+            SleepCheck(10), SleepCheck(10),
+            KeyboardInterruptCheck(), SleepCheck(10)
```

Comment:
Which unit test was failing without this change and how?

Reply:
The unittest was failing is `test_kbd_interrupt_in_wait_with_limited_concurrency`. This is because we set the max jobs allowed to 2 and the order of the processing of the tests is different. Refrane would pick up two `SleepCheck` tests, and when they are over it picks up the other `SleepCheck` and `KeyboardInterruptCheck`. Finally only one would be aborted this way.
This is the error before changing:
```
=============================================================================== FAILURES ================================================================================
__________________________________________________________ test_kbd_interrupt_in_wait_with_limited_concurrency __________________________________________________________

async_runner = (<reframe.frontend.executors.Runner object at 0x7fce9e40fa90>, <unittests.test_policies._TaskEventMonitor object at 0x7fce9e40f4d0>)
make_cases = <function make_cases.<locals>._make_cases at 0x7fce9e4c8e60>, make_exec_ctx = <function make_exec_ctx.<locals>._make_exec_ctx at 0x7fce9e5a7290>

    def test_kbd_interrupt_in_wait_with_limited_concurrency(
            async_runner, make_cases, make_exec_ctx
    ):
        # The general idea for this test is to allow enough time for all the
        # four checks to be submitted and at the same time we need the
        # KeyboardInterruptCheck to finish first (the corresponding wait should
        # trigger the failure), so as to make the framework kill the remaining
        # three.
        make_exec_ctx(options=max_jobs_opts(2))
        runner, _ = async_runner
        with pytest.raises(KeyboardInterrupt):
            runner.runall(make_cases([
                KeyboardInterruptCheck(), SleepCheck(10),
                SleepCheck(10), SleepCheck(10)
            ]))
    
>       assert_interrupted_run(runner)

unittests/test_policies.py:735: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

runner = <reframe.frontend.executors.Runner object at 0x7fce9e40fa90>

    def assert_interrupted_run(runner):
        assert 4 == runner.stats.num_cases()
        assert_runall(runner)
        assert 1 == len(runner.stats.failed())
>       assert 3 == len(runner.stats.aborted())
E       assert 3 == 1
E        +  where 1 = len([<reframe.frontend.executors.RegressionTask object at 0x7fce9e1d6750>])
E        +    where [<reframe.frontend.executors.RegressionTask object at 0x7fce9e1d6750>] = <bound method TestStats.aborted of <reframe.frontend.statistics.TestStats object at 0x7fce9e4186d0>>()
E        +      where <bound method TestStats.aborted of <reframe.frontend.statistics.TestStats object at 0x7fce9e4186d0>> = <reframe.frontend.statistics.TestStats object at 0x7fce9e4186d0>.aborted
E        +        where <reframe.frontend.statistics.TestStats object at 0x7fce9e4186d0> = <reframe.frontend.executors.Runner object at 0x7fce9e40fa90>.stats

unittests/test_policies.py:695: AssertionError
======================================================================== short test summary info ========================================================================
FAILED unittests/test_policies.py::test_kbd_interrupt_in_wait_with_limited_concurrency - assert 3 == 1
========================================================================== 1 failed in 12.72s ===========================================================================
```

Reply:
I think that this happened because you removed the logic of submitting tests from the the `runcase()`.